fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultDimensionConfig.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.dimension;
+
+import com.yahoo.bard.webservice.data.config.names.DimensionName;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+
+import java.util.LinkedHashSet;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * A Dimension instance holds all of the information needed to construct a Dimension.
+ */
+public class DefaultDimensionConfig implements DimensionConfig {
```

Comment:
I think the point of this change is so that an app-specific class isn't _needed_, so making this abstract defeats that purpose. This work originated from some folks standing up a new Fili instance that had also done a previous one, and noticing that they had to implemented the exact same class for each instance in order to implement the `DimensionConfig` interface.

Can you call out a specific reason for wanting this abstract? What additions would you expect a concrete version of that abstract one to add?